### PR TITLE
Nested routes

### PR DESCRIPTION
**Pull request by josephsavona**
Adds support for nested routes with preloading: all elements of the route load in parallel and by default there's just one`<Suspense>` boundary at the top. But individual items in the route path can wrap their  children in `<Suspense>`, in order to show parent components w/o waiting for children.

### CODE REVIEW

File: issue-tracker/src/routes.js [hunk 1]
```diff
@@ -39,25 +22,64 @@ const routes = [
...
-        issueDetailQuery: preloadQuery(
-          RelayEnvironment,
-          IssueDetailQuery,
-          {
-            id: params.id,
-            name: 'relay',
-            owner: 'facebook',
-          },
-          { fetchPolicy: 'store-or-network' },
+    routes: [
```
Comment (jstejada) [inline issue-tracker/src/routes.js]: how does this part work? is this react-router specific? at which point to we prepare the `Root` entry?

Comment (josephsavona) [reply issue-tracker/src/routes.js]: This is the format expected by react-router-config. You can have nested routes with or without paths (afaict), in this case i just want a single shared layout component so i don't need to provide a specific path for it. createRouter.js handles matching the current location to the *set* of matching nested routes and preloading all of them in parallel. Then <RouterRenderer> renders those items.

File: issue-tracker/src/routing/createRouter.js [hunk 2]
```diff
@@ -70,20 +65,23 @@ export default function createRouter(routes, options) {
...
+function prepareMatches(matches) {
+  return matches.map(match => {
+    const { route, match: matchData } = match;
+    const prepared = route.prepare(matchData.params);
+    const Component = route.component.get();
+    if (Component == null) {
+      route.component.load(); // eagerly load
+    }
+    return { component: route.component, prepared, routeData: matchData };
+  });
```
Comment (josephsavona) [inline issue-tracker/src/routing/createRouter.js]: @jstejada here's where we map each matching (nested) route entry to preload its code and data.

File: issue-tracker/src/routing/RouteRenderer.js [hunk 3]
```diff
@@ -51,6 +51,48 @@ export default function RouterRenderer() {
...
+      <RouteComponent
+        component={nextItem.component}
+        prepared={nextItem.prepared}
+        routeData={nextItem.routeData}
+      >
+        {routeComponent}
+      </RouteComponent>
+    );
+  }
+
```
Comment (josephsavona) [inline issue-tracker/src/routing/RouteRenderer.js]: and then here's where we render the already-preloaded route items

**Pull request merged**